cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java

```diff
@@ -97,6 +105,12 @@ public void launch(Job job) throws Exception {
     }
     systemArgs.put(ProgramOptionConstants.SCHEDULE_NAME, schedule.getName());
     systemArgs.put(ProgramOptionConstants.EVENT_NOTIFICATIONS, GSON.toJson(job.getNotifications()));
+    SatisfiableTrigger trigger = ((SatisfiableTrigger) job.getSchedule().getTrigger());
+    TriggerInfoContext triggerInfoContext = new TriggerInfoContext(job, store);
```

Comment:
Ideally in here you just create the `TriggeringScheduleInfo` (via a private method). That method can further divided, but it's not the concern in here (programming by intention).